addon-test-support/RouteInfoMock.ts

```diff
@@ -0,0 +1,75 @@
+interface RouteInfo {
+  readonly name: string;
+  readonly parent: RouteInfo | null;
+  readonly child: RouteInfo | null;
+  readonly localName: string;
+  readonly params: { [name: string]: unknown } | undefined;
+  readonly paramNames: string[];
+  readonly queryParams: { [name: string]: string };
+  readonly metadata: unknown;
+  find(
+    predicate: (this: any, routeInfo: RouteInfo, i: number) => boolean,
+    thisArg?: any
+  ): RouteInfo | undefined;
+}
+
+export default class RouteInfoMock implements RouteInfo {
+  name: string;
+  parent: RouteInfo | null = null;
+  child: RouteInfo | null = null;
+  localName: string;
+  params: { [name: string]: unknown } = {};
+  paramNames: string[] = [];
+  queryParams: { [name: string]: string } = {};
+  metadata: unknown;
+
+  constructor(name: string) {
+    this.name = name;
+    this.localName = name.split('.').pop()!;
+  }
+
+  withParent(parent: RouteInfo): RouteInfoMock {
+    this.parent = parent;
+    return this;
+  }
+
+  withChild(child: RouteInfo): RouteInfoMock {
+    this.child = child;
+    return this;
+  }
+
+  withParams(params: { [name: string]: unknown }): RouteInfoMock {
```

Comment:
This also needs updating for `params`' type:

```suggestion
  withParams(params: Record<string, string>): RouteInfoMock {
```